### PR TITLE
Make it easier to honor --quiet

### PR DESCRIPTION
**Pull request by jsoref**
This should help address #4163.

This doesn't cover passing quiet into the various platform bits, but at least it should make certbot-auto itself quieter...

### CODE REVIEW

File: letsencrypt-auto-source/pieces/bootstrappers/deb_common.sh [hunk 1]
```diff
@@ -21,7 +21,7 @@ BootstrapDebCommon() {
     QUIET_FLAG='-qq'
   fi
 
-  $SUDO apt-get $QUIET_FLAG update || echo apt-get update hit problems but continuing anyway...
+  $SUDO apt-get $QUIET_FLAG update || say apt-get update hit problems but continuing anyway...
```
Comment (bmw) [inline letsencrypt-auto-source/pieces/bootstrappers/deb_common.sh]: I think this should be `error` so the user is notified that we encountered problems.

File: letsencrypt-auto-source/pieces/bootstrappers/rpm_common.sh [hunk 2]
```diff
@@ -25,9 +25,9 @@ BootstrapRpmCommon() {
   fi
 
   if ! $SUDO $tool list *virtualenv >/dev/null 2>&1; then
-    echo "To use Certbot, packages from the EPEL repository need to be installed."
+    say "To use Certbot, packages from the EPEL repository need to be installed."
```
Comment (bmw) [inline letsencrypt-auto-source/pieces/bootstrappers/rpm_common.sh]: I'm a little unsure we should be using `say` here as well. This code should only be executed the first time `certbot-auto` runs on a system. In this case, if the user runs `certbot-auto --quiet`, they'll get a `sudo` prompt with no context. I'm not sure if this is the behavior we want.

File: letsencrypt-auto-source/letsencrypt-auto.template [hunk 3]
```diff
@@ -309,42 +319,40 @@ UNLIKELY_EOF
...
+    say "Installation succeeded."
   fi
   if [ -n "$SUDO" ]; then
     # SUDO is su wrapper or sudo
-    if [  "$QUIET" != 1 ]; then
-      echo "Requesting root privileges to run certbot..."
-      echo "  $VENV_BIN/letsencrypt" "$@"
-    fi
+    say "Requesting root privileges to run certbot..."
+    say "  $VENV_BIN/letsencrypt" "$@"
```
Comment (bmw) [inline letsencrypt-auto-source/letsencrypt-auto.template]: I have the same comment here about the promptless `sudo` call when run interactively. Unless you have thoughts about how to work around this, I recommend leaving these as `echo` for now and we'll continue work on this.

File: letsencrypt-auto-source/letsencrypt-auto.template [hunk 4]
```diff
@@ -385,9 +393,9 @@ UNLIKELY_EOF
     # ---------------------------------------------------------------------------
     DeterminePythonVersion
     if ! REMOTE_VERSION=`"$LE_PYTHON" "$TEMP_DIR/fetch.py" --latest-version` ; then
-      echo "WARNING: unable to check for updates."
+      say "WARNING: unable to check for updates."
```
Comment (bmw) [inline letsencrypt-auto-source/letsencrypt-auto.template]: I think this should be `error` as well so the user is notified `certbot-auto` encountered a problem.

File: letsencrypt-auto-source/pieces/bootstrappers/rpm_common.sh (hunk 2 above)
Comment (jsoref) [inline letsencrypt-auto-source/pieces/bootstrappers/rpm_common.sh]: Well, we might be able to do `sudo -A ...` and use the `-A` argument to spit out stuff we should have printed earlier... I'm not sure how portable `-A` is...

Comment (jsoref) [inline letsencrypt-auto-source/pieces/bootstrappers/rpm_common.sh]: I think we could probably try running `$tool list ...`, capture and process the output and then decide to spit out the message and run `sudo $tool ...`.

Comment (bmw) [inline letsencrypt-auto-source/pieces/bootstrappers/rpm_common.sh]: We could also wrap `sudo` the way we do for `su` to echo the command we're about to run, but many users only have to type in their `sudo` password the first time meaning we'll be printing a lot of unnecessary output. Ideally we find a way to only print the message if we know we're going to prompt for a password. We could do this by capturing the output, but I want to make sure whatever we do isn't too complex and is robust on the wide variety of systems `certbot-auto` is run on.

**Pull request merged**